src/fe/fe_interface.C

```diff
@@ -680,6 +680,8 @@ void FEInterface::shape<Real>(const unsigned int dim,
   if (is_InfFE_elem(t))
     phi = ifem_shape(dim, fe_t, t, i, p);
 
+  // here we should need an 'or'-statement, right? Else, we will run in trouble!?
+
```

Comment:
Braces and a "return", I'd say.

This function IIRC comes from @pbauman's refactoring for vector-valued elements, which was kind of in the Dark Ages for infinite elements.  I would be ashamed but unsurprised if we failed to get infinite element support correct here and nobody realized it.